src/Stripe.net/Infrastructure/JsonConverters/STJAnyOfConverter.cs

```diff
@@ -0,0 +1,96 @@
+#if NET6_0_OR_GREATER
+namespace Stripe.Infrastructure
+{
+    using System;
+    using System.Linq;
+    using System.Reflection;
+    using System.Text.Json;
+    using System.Text.Json.Serialization;
+
+    /// <summary>
+    /// Converts a <see cref="IAnyOf"/> to and from JSON.
+    /// </summary>
+    public class STJAnyOfConverter : JsonConverter<object>
+    {
+        /// <summary>
+        /// Reads the JSON representation of the object.
+        /// </summary>
+        /// <param name="reader">The <see cref="Utf8JsonReader"/> to read from.</param>
+        /// <param name="typeToConvert">Type of the object.</param>
+        /// <param name="options">The calling serializer's options.</param>
+        /// <returns>The object value.</returns>
+        public override object Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType == JsonTokenType.Null)
+            {
+                return null;
+            }
+
+            object o = null;
+
+            // Try to deserialize with each possible type
+            var elem = JsonElement.ParseValue(ref reader);
+            foreach (var type in typeToConvert.GenericTypeArguments)
+            {
+                try
+                {
+                    o = JsonSerializer.Deserialize(elem.GetRawText(), type, options);
```

Comment:
do we have to go through raw text here?

Reply:
Same as with STJStripeObjectConverter; I think this is closely aligned with what we do in NS, but we may be able to make this better.

Reply:
Update: there are overrides that I didn't see initially that accept the JsonElement, so we don't need to go thru raw text.